dianna/methods/shap.py

```diff
@@ -1,21 +1,223 @@
+
+    def explain_image(
+        self,
+        model,
+        input_data,
+        nsamples="auto",
+        background=None,
+        n_segments=100,
+        compactness=10.0,
+        sigma=0,
+        channel_axis_first=False,
+        **kwargs,
+    ):
+        """Run the KernelSHAP explainer.
+           The model will be called with the function of image segmentation.
+
+        Args:
+            model (str): The path to a ONNX model on disk.
+            input_data (np.ndarray): Data to be explained. It is mandatory to only
+                                     provide a single example as input. This is because
+                                     LIME / KernelShap is generally used for sample-based
+                                     interpretability, training a separate interpretable
+                                     model to explain a model’s prediction on each individual
+                                     example. The input dimension must be
+                                     [height, width, color_channels] or
+                                     [color_channels, height, width] (see channel_axis_first)
+            nsamples ("auto" or int): Number of times to re-evaluate the model when
+                                      explaining each prediction. More samples lead
+                                      to lower variance estimates of the SHAP values.
+                                      The "auto" setting uses
+                                      `nsamples = 2 * X.shape[1] + 2048`
+            background (int): Background color for the masked image
+            n_segments (int): The (approximate) number of labels in the segmented output image
+            compactness (int): Balances color proximity and space proximity. Higher values give
+                               more weight to space proximity, making superpixel shapes more
+                               square/cubic.
+            sigma (float): Width of Gaussian smoothing kernel for pre-processing for
+                           each dimension of the image. Zero means no smoothing.
+            channel_axis_first (boolean): If True, the shape of the input image is
+                                          [color_channels, height, width]
+
+        Other keyword arguments: see the documentation of kernel explainer of SHAP
+                                 (also in function "shap_values") via:
+        https://github.com/slundberg/shap/blob/master/shap/explainers/_kernel.py
+                                 and the documentation of image segmentation via:
+        https://scikit-image.org/docs/dev/api/skimage.segmentation.html#skimage.segmentation.slic
+
+        Returns:
+            Explanation heatmap of shapley values for each class (np.ndarray).
         """
-        Example call function.
+        self.onnx_model = onnx.load(model)  # load onnx model
+        self.output_node = prepare(self.onnx_model, gen_tensor_dict=True).outputs[0]
```

Comment:
Would it make sense to move the onnx to tf conversion to dianna.utils?

Reply:
Thanks for your comment @loostrum . Sounds good to me! I will move them to utils to make the code cleaner.